target/linux/octeon/base-files/lib/upgrade/platform.sh

```diff
@@ -37,21 +41,39 @@ platform_do_flash() {
 	local kernel=$3
 	local rootfs=$4
 
+	local board_dir=$(tar tf "$1" | grep -m 1 '^sysupgrade-.*/$')
+		board_dir=${board_dir%/}
+	[ -n "$board_dir" ] || return
+
 	mkdir -p /boot
-	mount -t vfat /dev/$kernel /boot
-
-	[ -f /boot/vmlinux.64 -a ! -L /boot/vmlinux.64 ] && {
-		mv /boot/vmlinux.64 /boot/vmlinux.64.previous
-		mv /boot/vmlinux.64.md5 /boot/vmlinux.64.md5.previous
-	}
-
-	echo "flashing kernel to /dev/$kernel"
-	tar xf $tar_file sysupgrade-$board/kernel -O > /boot/vmlinux.64
-	md5sum /boot/vmlinux.64 | cut -f1 -d " " > /boot/vmlinux.64.md5
-	echo "flashing rootfs to ${rootfs}"
-	tar xf $tar_file sysupgrade-$board/root -O | dd of="${rootfs}" bs=4096
+
+	if [ $board = "itus,shield-router" ]; then
+		# mmcblk1p1 (fat) contains all ELF-bin images for the Shield
+		mount /dev/mmcblk1p1 /boot
+		echo "flashing Itus Kernel to /boot/$kernel (/dev/mmblk1p1)"
+		tar -C /tmp -xvf $tar_file
+		cp "/tmp/$board_dir/kernel" /boot/$kernel
+		umount /boot
+		echo "flashing rootfs to ${rootfs}"
+		dd if="/tmp/$board_dir/root" of="${rootfs}"
+	else
+		echo "flashing kernel to /dev/$kernel"
+		mount -t vfat /dev/$kernel /boot
+
+		[ -f /boot/vmlinux.64 -a ! -L /boot/vmlinux.64 ] && {
+			mv /boot/vmlinux.64 /boot/vmlinux.64.previous
+			mv /boot/vmlinux.64.md5 /boot/vmlinux.64.md5.previous
+		}
+
+		echo "flashing kernel to /dev/$kernel"
+		tar xf $tar_file $board_dir/kernel -O > /boot/vmlinux.64
+		md5sum /boot/vmlinux.64 | cut -f1 -d " " > /boot/vmlinux.64.md5
+		umount /boot
+
+		echo "flashing rootfs to ${rootfs}"
+		tar xvf $tar_file $board_dir/root -O | dd of="${rootfs}" bs=4096
```

Comment:
Same here, the v hasn't been there before.

Reply:
Removed the verbose tag on tar